### PR TITLE
ENH/TST: add spiral and fermat spiral scans

### PR DESCRIPTION
**Pull request by klauer**
This was a lot simpler when it was hidden in hxntools...

### CODE REVIEW

**Comment by tacaswell** [comment]
Why was it simpler hidden in hxntools?

File: bluesky/spec_api.py [hunk 1]
```diff
@@ -357,3 +355,162 @@ def tw(motor, step, time=None, *, md=None):
...
+        any 'setable' object (motor, temp controller, etc.)
+    y_motor : object
+        any 'setable' object (motor, temp controller, etc.)
+    x_range : float
+        X range, in engineering units
+    y_range : float
+        Y range, in engineering units
+    dr : float
+        Delta radius, in engineering units
+    nth : float
```
Comment (tacaswell) [inline bluesky/spec_api.py]: nth is not passed through?

**Comment by klauer** [comment]
@tacaswell no class wrappers, no 'spec api'/gs-aware separation, no tests... and no PR means no bikeshedding 😄

**Comment by tacaswell** [comment]
We could just drop the spec api / gs....

File: bluesky/__init__.py [hunk 2]
```diff
@@ -24,6 +24,10 @@
 from .plans import OuterProductAbsScanPlan
 from .plans import OuterProductDeltaScanPlan
 from .plans import Tweak
+from .plans import SpiralScan
```
Comment (danielballan) [inline bluesky/__init__.py]: The names imported here are the *old* names, imported for back-compat. (One of the PRs I'm working on adds a comment to this effect.) I don't think we need to import the new names at top-level, but if you feel differently I have no problem with it.

File: bluesky/plans.py [hunk 3]
```diff
@@ -1950,6 +1950,204 @@ def core():
...
+    per_step : callable, optional
+        hook for cutomizing action of inner loop (messages per step)
+        See docstring of bluesky.plans.one_nd_step (the default) for
+        details.
+    md : dict, optional
+        metadata
+
+    See Also
+    --------
+    `bluesky.plans.spiral`
```
Comment (danielballan) [inline bluesky/plans.py]: Nice. I see that "See Also" is about to sweep through our codebase.

File: bluesky/__init__.py (hunk 2 above)
Comment (klauer) [inline bluesky/__init__.py]: Removed

File: bluesky/spec_api.py (hunk 1 above)
Comment (klauer) [inline bluesky/spec_api.py]: I think you meant to leave this on the next function down - it was named factor, should have been nth. (correct me if I'm wrong)

**Comment by klauer** [comment]
@tacaswell up to you as far as dropping the gs-aware one.

**Comment by tacaswell** [comment]
I was being sarcastic about dropping gs

**Comment by tacaswell** [comment]
attn @gjwillms @imrealkaren

**Pull request merged**